From 1786e40ec996b3561ed2aea9e6673c5084c2c8c4 Mon Sep 17 00:00:00 2001
Message-Id: <patch-v2-18.33-1786e40ec99-20220831T230246Z-avarab@gmail.com>
In-Reply-To: <cover-v2-00.33-00000000000-20220831T230246Z-avarab@gmail.com>
References: <cover.1657819649.git.matheus.bernardino@usp.br>
	<cover-v2-00.33-00000000000-20220831T230246Z-avarab@gmail.com>
From: =?UTF-8?q?=C3=86var=20Arnfj=C3=B6r=C3=B0=20Bjarmason?=
 <avarab@gmail.com>
Date: Mon, 1 Aug 2022 17:33:17 +0200
Subject: [PATCH v2 18/33] submodule--helper: don't redundantly check "else if
 (res)"
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

The "res" variable must be true at this point in update_submodule(),
as just a few lines above this we've unconditionally:

	if (!res)
		return 0;

So we don't need to guard the "return 1" with an "else if (res)", we
can return unconditionally at this point. See b3c5f5cb048 (submodule:
move core cmd_update() logic to C, 2022-03-15) for the initial
introduction of this code, this check of "res" has always been
redundant.

Signed-off-by: Ævar Arnfjörð Bjarmason <avarab@gmail.com>
---
 builtin/submodule--helper.c | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/builtin/submodule--helper.c b/builtin/submodule--helper.c
index 07a918f7373..93ab49defb1 100644
--- a/builtin/submodule--helper.c
+++ b/builtin/submodule--helper.c
@@ -2448,8 +2448,7 @@ static int update_submodule(struct update_data *update_data)
 			    update_data->displaypath);
 		if (res == 128)
 			exit(res);
-		else if (res)
-			return 1;
+		return 1;
 	}
 
 	return 0;
-- 
2.37.3.1420.g76f8a3d556c

